test/Helper/HeadScriptTest.php

```diff
@@ -551,4 +554,43 @@ public function testOmitsTypeAttributeIfNoneGivenAndHtml5Doctype()
         $test = $this->helper->__invoke()->toString();
         $this->assertDoesNotMatchRegularExpression('#type="text/javascript"#i', $test);
     }
+
+    public function testSupportsNonceAttribute(): void
+    {
+        ($this->helper)()->appendScript(
+            '// some js',
+            'text/javascript',
+            ['nonce' => 'random']
+        );
+
+        self::assertStringContainsString(
+            'nonce="random"',
+            (string) ($this->helper)()
+        );
+    }
+
+    /** @return Generator<string, array<int, string> */
+    public function booleanAttributeDataProvider(): Generator
+    {
+        $values = ['async', 'defer', 'nomodule'];
+
+        foreach ($values as $name) {
+            yield $name => [$name];
+        }
+    }
+
+    /** @dataProvider booleanAttributeDataProvider */
+    public function testBooleanAttributesUseTheKeyNameAsTheValue(string $attribute): void
```

Comment:
If you add the support for attributes that behave as [booleans](https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#boolean-attribute) then this should work as well, based on the doctype:

```html
<script type="module" async></script>
```

Reply:
Hey @froschdesign - I'm not sure what you mean here. I've added a test to prove that boolean attributes can be assigned to script elements marked as modules. The style of the booleans is `attr="attr"` as opposed to `attr` - both are valid - are you suggesting to change the style to the latter?

Figuring out whether `nomodule` is mutually exclusive with a `module` type - do you mean that??

Reply:
We leave everything as it is, because with version 3 at the latest, the entire attribute creation must be revised. I don't like at all that the [abstract view helper in laminas-form](https://github.com/laminas/laminas-form/blob/c6a69c93e9c445b9a2e08cd7a4746bc724bd4ffe/src/View/Helper/AbstractHelper.php#L30) does something more or different.
And with #60 is something new in the pipeline.